### PR TITLE
Switch centipawn to logit and refactor score calculations.

### PR DESCRIPTION
**Pull request by Mardak**
r?@mooskagh From https://github.com/LeelaChessZero/lc0/pull/860#issuecomment-500215701 and pulling out `edge.GetQ(default_q)` from each branch and consistently clamp and round to ±12800 (for ±Infinity)

Differences summary before/after shown below:
centipawn: startpos 32 -> 29 (febc5980608326685fe0c7c839c3f90288cab060's new centipawn vs logit)
win_percentage: startpos 5543 -> 5544 (rounding instead of truncate)
others unchanged

before (febc5980608326685fe0c7c839c3f90288cab060):
```
# position startpos

centipawn: score cp 32
centipawn_2018: score cp 49
win_percentage: score cp 5543
Q: score cp 1087

# position fen r2qk2r/ppp2p2/3p4/2b1p1pp/2BnP1nP/3P1PB1/PPP3P1/RN1Q1RK1 b kq - 0 13 moves d4e2

centipawn: score cp -12800
centipawn_2018: score cp -12800
win_percentage: score cp 0
Q: score cp -10000

# position fen r2qk2r/ppp2p2/3p4/2b1p1pp/2BnP1nP/3P1PB1/PPP3P1/RN1Q1RK1 b kq - 0 13 moves d4e2 g1h1

centipawn: score cp 12800
centipawn_2018: score cp 12800
win_percentage: score cp 10000
Q: score cp 10000
```
after:
```
# position startpos

centipawn: score cp 29
centipawn_2018: score cp 49
win_percentage: score cp 5544
Q: score cp 1087

# position fen r2qk2r/ppp2p2/3p4/2b1p1pp/2BnP1nP/3P1PB1/PPP3P1/RN1Q1RK1 b kq - 0 13 moves d4e2

centipawn: score cp -12800
centipawn_2018: score cp -12800
win_percentage: score cp 0
Q: score cp -10000

# position fen r2qk2r/ppp2p2/3p4/2b1p1pp/2BnP1nP/3P1PB1/PPP3P1/RN1Q1RK1 b kq - 0 13 moves d4e2 g1h1

centipawn: score cp 12800
centipawn_2018: score cp 12800
win_percentage: score cp 10000
Q: score cp 10000
```

### CODE REVIEW

**Comment by Mardak** [comment]
Simplified logit math converting `log(p/(1-p))` to `-log(1/p-1)` where `p = Q/2+.5 = (Q+1)/2`, so `1/p = 2/(Q+1)`

**Comment by Mardak** [comment]
Also, the cp to winpct conversion is straightforward:

```
# javascript
win = 1 / (1 + Math.pow(Math.E, -cp / 135))

# python
import math
win = 1 / (1 + math.exp(-cp / 135.0))
```

**Comment by AlexisOlson** [comment]
I think if PR925 is implemented, this will make even more sense. If the logit Q gets applied in training, I'd expect the extreme Q evals would get reshaped somewhat and logit would give a better fit than it did when #860 was merged.

File: src/mcts/search.cc [hunk 1]
```diff
@@ -125,16 +125,17 @@ void Search::SendUciInfo() REQUIRES(nodes_mutex_) {
     ++multipv;
     uci_infos.emplace_back(common_info);
     auto& uci_info = uci_infos.back();
+    auto score = edge.GetQ(default_q);
     if (score_type == "centipawn") {
-      uci_info.score = 295 * edge.GetQ(default_q) /
-                       (1 - 0.976953126 * std::pow(edge.GetQ(default_q), 14));
+      score = 135 * -std::log(2 / (score + 1) - 1);
```
Comment (mooskagh) [inline src/mcts/search.cc]: Maybe keep also the current way under some name? `centipawn_2019`, for example.

**Comment by mooskagh** [comment]
Would be nice to plot graphs of before/after.

File: src/mcts/search.cc (hunk 1 above)
Comment (Mardak) [reply src/mcts/search.cc]: Restored and refactored the current way.

**Comment by Mardak** [comment]
@zz4032 provided a plot with logit in https://github.com/LeelaChessZero/lc0/pull/860#issuecomment-500215701

Rebased and added `centipawn_2019`

**Comment by mooskagh** [comment]
As discussed in Discord, it seems that this change makes eval almost impossible to reach 10 pawns (TCEC threshold for adjudication). Which some people may think as an issue.

**Comment by Mardak** [comment]
@zz4032 any thoughts on if the formula should be adjusted? The SF blue dots do seem like they would exceed 10 pawn only for very extreme Q and logit does seem to match that behavior.

**Comment by zz4032** [comment]
@Mardak The logit function is more on the outer side of most of the blue SF dots in the Q-range close to Q=1. No idea currently how to modify function to be closer to the current (orange) curve here.

**Comment by Mardak** [comment]
I'll close this for now as there hasn't been a suggestion on how to avoid issues raised in https://github.com/LeelaChessZero/lc0/pull/881#issuecomment-534245922

**Pull request closed without merging**